Create 06 smallestWindow.cpp

Juro221 opened the pull request:
First we fill our our map with elements of string P, and we update thier frequencies. Then we iterate thorugh our string S. We are looking for elements that are in our unordered map. When we find all our elements that we wanted to find we need to check if there is not a smaller answer. We do that again by some form of window sliding.
Time complexity : O(|S|)

JurajZitnansky committed:
Create 06 smallestWindow.cpp

First we fill our our map with elements of string P, and we update thier frequencies. Then we iterate thorugh our string S. We are looking for elements that are in our unordered map. When we find all our elements that we wanted to find we need to check if there is not a smaller answer. We do that again by some form of window sliding.
Time complexity : O(|S|)

FazeelUsmani reviewed (commented):
Getting TLE.
Try to optimize

JurajZitnansky committed:
Update 06 smallestWindow.cpp

It seems like the unordered_map was too slow so i replaced it. Hopefully this wont cause TLE.
And the time complexity will be O(|S|)

FazeelUsmani reviewed (approved):
Thank you for contributing
